src/domain/openvino_models.py

```diff
@@ -0,0 +1,60 @@
+from PytorchWildlife.models import detection as pw_detection
+import openvino.properties as props
+import openvino as ov
+import numpy as np
+import torch
+
+core = ov.Core()
+core.set_property({props.cache_dir: "cache"})
+#TODO: Add NPU specific properties
+
+class DetectionModel():
+    def __init__(self, device="cpu") -> None:
+        print("Loading Openvino model")
+        
+        ov_model = core.read_model("detection_model.onnx")
+        self.model = ov.compile_model(ov_model, device_name=device.upper())
+
+    def __call__(self, img: torch.tensor):
+       results = [torch.tensor(t) for t in self.model(img).values()]
```

Comment:
It could be:

       return [torch.tensor(t) for t in self.model(img).values()]

However, is it really needed to give back a list? If not strictly required, an iterator could be a more efficient solution (for example, because the values are just used in a for loop):

       return (torch.tensor(t) for t in self.model(img).values())

Reply:
@alessandropalla do you want to handle this?